### PR TITLE
adding getFactory

### PR DESCRIPTION
**Pull request by hugo-gomes**
---
related to issue: https://github.com/ripe-tech/ripe-core/issues/4194
description: adding getFactory
---

### CODE REVIEW

**Comment by joamag** [comment]
@hugo-gomes please cehck the unit tests

File: test/js/api.js [hunk 1]
```diff
@@ -117,11 +117,11 @@ describe("RipeAPI", function() {
             assert.notStrictEqual(typeof result.sid, undefined);
 
             result = await new Promise((resolve, reject) => {
-                remote.getOrder(853, resolve);
+                remote.getOrder(4488, resolve);
             });
 
-            assert.strictEqual(result.number, 853);
-            assert.strictEqual(result.number_s, "#000853");
+            assert.strictEqual(result.number, 4488);
```
Comment (joamag) [inline test/js/api.js]: we'll need to find a more future proof solution latter :)

File: src/python/ripe_demo/static/js/ripe.js [hunk 2]
```diff
@@ -1092,6 +1100,18 @@ ripe.Ripe.prototype._getCombinationsOptions = function(options) {
...
 
+ripe.Ripe.prototype._getFactoryOptions = function(options) {
+    options = options || {};
+    var brand = options.brand === undefined ? this.brand : options.brand;
+    var model = options.model === undefined ? this.model : options.model;
+    var url = this.url + "brands/" + brand + "/models/" + model + "/factory";
+    return Object.assign(options, {
+        url: url,
+        method: "GET",
+        params: {}
```
Comment (joamag) [inline src/python/ripe_demo/static/js/ripe.js]: remove this params option (not used)

File: test/js/api.js (hunk 1 above)
Comment (hugo-gomes) [reply test/js/api.js]: discussed this @acaldas a couple of minutes ago :)

**Comment by joamag** [comment]
@hugo-gomes please run the build process

**Pull request merged**